shell/components/auth/SelectPrincipal.vue

```diff
@@ -41,6 +41,11 @@ export default {
       type:    Boolean,
       default: false
     },
+
+    showTooltip: {
```

Comment:
Overall the behavior of the tooltip looks good to me, but I think it can be applied to almost all uses of SelectPrincipal because they (almost all) have the same issue as the cluster membership editor. I think whether or not to show the tooltip can be determined by the `retainSelection` prop instead of a new prop. This would enable the tooltip for project membership editing and group role assignment.


 When `retainSelection` is `false` the selected principal is not shown inside the select component and name length looks like a non-issue to me; there's a lot of vertical space in the form for the principal name to wrap around

![Screen Shot 2022-12-15 at 1 39 14 PM](https://user-images.githubusercontent.com/42977925/207962337-97ac7a76-c80a-4939-b381-59559524d318.png)

Reply:
Make sense to me! thanks.